build-tools/packages/build-infrastructure/src/fluidRepo.ts

```diff
@@ -235,3 +237,30 @@ export function getAllDependenciesInRepo(
 		workspaces: [...workspaces],
 	};
 }
+
+export async function setDependencyRange(
+	packagesToUpdate: IPackage[],
+	dependencies: IPackage[],
+	dependencyRange: InterdependencyRange
+): Promise<void> {
+	for (const pkg of packagesToUpdate) {
+		for (const dep of pkg.combinedDependencies) {
```

Comment:
combinedDependencies returns an object with the dep name and kind, so you can do something like this:

```suggestion
		for (const {name: depName, depClass} of pkg.combinedDependencies) {
```

Which will simplify the code below a bit, since you'll know which property in package.json to update (i.e. deps, devDeps, or peerDeps).